pysd/py_backend/builder.py

```diff
@@ -771,7 +768,7 @@ def merge_partial_elements(element_list):
     return list(outs.values())
 
 
-def add_stock(identifier, expression, initial_condition, subs):
+def add_stock(identifier, expression, initial_condition, subs, merge_subs):
```

Comment:
Hey @enekomartinmartinez - just got back from a bunch of travel and I'm a bit tired, so I may be missing things. Can you help me understand more about what `merge_subs` means here and why it differs from just `subs`? It may help to expand the docstring to give a bit more detail about how this changes the behavior of the function. =)

Reply:
Actually, I think I understand more in reading the rest of the code. This is for cases where multiple subscript ranges are defined and then PySD is combining them for its internal operation, correct?

Reply:
Yes, before I added `make_merge_list` a long time ago, we only can merge safely objects where the dimensions only appear in one subscript range.

I added this function to be able to use models with repeated subscripts in subscripts subranges, e.g.:
```
dim1: A, B, C ~~|
dim2: A, B, C ~~|
```
or with subranges:
```
dim1: A,B,C ~~|
dimi: A,B ~~|
dimf: B,C ~~|
```

When I first added the function I did it directly when doing the merge in the builder (when `xrmerge` function is added). However, in order to improve the speed and reduce the number of calls, I noticed that having this information when visiting common expressions in the parser was useful. This way we could avoid calling `rearrange` when the coordinates of the object in a reference are the same as the output from the merge. This information is given through `elements_subs_dict`. This forced me to call `make_merge_list` before in the vensim2py before parsing general expressions.

Then, I was calling the function twice, but that didn't make a lot of sense, so while doing this PR I decided to save that information with the `merge_subs` key, which in most of the cases will be the same value as `subs`. This allows only calling `make_merge_list` once. However, I need to pass this new information to the stateful objects builders and external builders (I also needed to pass the information to external builders to give them the correct dimension name).

Maybe the best way to see how it works is to take a look at the test cases in _tests/unit_tests_utils.py:test_make_merge_list_ .

Reply:
That makes sense. Thanks for explaining it to me. =)